vllm/entrypoints/llm.py

```diff
@@ -351,15 +351,16 @@ def generate(
 
     def chat(
         self,
-        messages: List[ChatCompletionMessageParam],
+        conversations: Union[List[ChatCompletionMessageParam],
+                             List[List[ChatCompletionMessageParam]]],
         sampling_params: Optional[Union[SamplingParams,
                                         List[SamplingParams]]] = None,
         use_tqdm: bool = True,
         lora_request: Optional[LoRARequest] = None,
         chat_template: Optional[str] = None,
         add_generation_prompt: bool = True,
         tools: Optional[List[Dict[str, Any]]] = None,
-    ) -> List[RequestOutput]:
+    ) -> Union[List[List[RequestOutput]], List[RequestOutput]]:
```

Comment:
Let's use overloads to better indicate the return type based on the type of `conversations`